bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieStateManager.java

```diff
@@ -61,15 +67,47 @@
     private final String bookieId;
     private ShutdownHandler shutdownHandler;
     private RegistrationManager registrationManager;
+    // Expose Stats
+    private final StatsLogger statsLogger;
 
 
-    public BookieStateManager(ServerConfiguration conf, RegistrationManager registrationManager,
-                              LedgerDirsManager ledgerDirsManager) throws IOException {
+    public BookieStateManager(ServerConfiguration conf, StatsLogger statsLogger,
+           RegistrationManager registrationManager, LedgerDirsManager ledgerDirsManager) throws IOException {
         this.conf = conf;
+        this.statsLogger = statsLogger;
         this.registrationManager = registrationManager;
         this.ledgerDirsManager = ledgerDirsManager;
         // ZK ephemeral node for this Bookie.
         this.bookieId = getMyId();
+        // 1 : up, 0 : readonly, -1 : unregistered
+        statsLogger.registerGauge(SERVER_STATUS, new Gauge<Number>() {
+            @Override
+            public Number getDefaultValue() {
+                return 0;
+            }
+
+            @Override
+            public Number getSample() {
+                if (!rmRegistered.get()){
+                    return -1;
+                } else if (forceReadOnly.get() || bookieStatus.isInReadOnlyMode()) {
+                    return 0;
+                } else {
+                    return 1;
+                }
+            }
+        });
+    }
+
+    @VisibleForTesting
+    BookieStateManager(ServerConfiguration conf, RegistrationManager registrationManager) throws IOException {
+        this.conf = conf;
+        this.registrationManager = registrationManager;
+        this.statsLogger = null;
```

Comment:
use NullStatsLogger.INSTANCE rather than plain `null`

Also I think this constructor use this other constructor

```
this(conf, NullStatsLogger.INSTANCE, registrationManager, new LedgerDirsManager(...));
```